src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs

```diff
@@ -146,15 +146,20 @@ sealed class AdoConnector {
             }
         }
 
-        public async IAsyncEnumerable<WorkItem> ExistingWorkItems() {
+        public async IAsyncEnumerable<WorkItem> ExistingWorkItems((string, string)[] notificationInfo) {
             var filters = new Dictionary<string, string>();
             foreach (var key in _config.UniqueFields) {
                 var filter = string.Empty;
                 if (string.Equals("System.TeamProject", key)) {
                     filter = await Render(_config.Project);
                 } else {
-                    filter = await Render(_config.AdoFields[key]);
+                    try {
+                        filter = await Render(_config.AdoFields[key]);
+                    } catch (Exception e) {
```

Comment:
is there a more specific exception type we can use here ?

Reply:
Or alternatively:

```csharp
if (string.Equals("System.TeamProject", key)) {
   filter = await Render(_config.Project);
} else if (_config.AdoFields.ContainsKey(key)) {
   filter = await Render(_config.AdoFields[key]);
} else {
   _logTracer.WithTags(notificationInfo).Error($"Failed to check for existing work items using the UniqueField Key: {key}. Value is not present in config field AdoFields.");
   continue;
}
```

The `continue` in the else branch is important because with the PRs current state, we would still be adding a filter for the erroring key but the filter would be empty.